app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 class PostsController < ApplicationController
+  helper_method :current_user
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.includes(comments: :author)
+    @posts = @user.posts.includes(comments: :user)
   end
 
   def show
-    @post = Post.includes(comments: :author).find(params[:id])
-    @user = @post.author
+    @user = User.find(params[:user_id])
+    @post = @user.posts.includes(comments: :user).find(params[:id])
+    @new_like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
```

Comment:
[OPTIONAL] Finding the user is not necessary since you are using `current_user`

Reply:
👌